internal/clock/simulated_clock_test.go

```diff
@@ -0,0 +1,279 @@
+		afterDuration time.Duration
+		action        func(sc *SimulatedClock) // Action to manipulate the clock after After() is called
+	}{
+		{
+			name:          "ZeroDuration_FiresImmediately",
+			afterDuration: 0,
+			action:        func(sc *SimulatedClock) { /* No action needed for immediate fire */ },
+		},
+		{
+			name:          "NegativeDuration_FiresImmediately",
+			afterDuration: -5 * time.Second,
+			action:        func(sc *SimulatedClock) { /* No action needed for immediate fire */ },
+		},
+		{
+			name:          "PositiveDuration_Fires_WhenTimeAdvancedPastDuration",
+			afterDuration: 10 * time.Second,
+			action: func(sc *SimulatedClock) {
+				sc.AdvanceTime(15 * time.Second) // Advance well past the duration
+			},
+		},
+		{
+			name:          "PositiveDuration_Fires_WhenTimeSetPastDuration",
+			afterDuration: 10 * time.Second,
+			action: func(sc *SimulatedClock) {
+				sc.SetTime(referenceTime.Add(15 * time.Second)) // Set time well past the duration
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			clock := NewSimulatedClock(referenceTime)
+			clockTimeAtAfterCall := clock.Now()
+
+			ch := clock.After(tc.afterDuration)
+			require.NotNil(t, ch, "Channel should not be nil")
+
+			// Perform the action (if any) that might trigger the timer
+			if tc.action != nil {
+				tc.action(clock)
+			}
+
+			// Determine the expected time on the channel if it fires
+			// For zero/negative duration, it's the time After() was called.
+			// For positive, it's that time + duration.
+			var expectedFireTimeOnChannel time.Time
+			if tc.afterDuration <= 0 {
+				expectedFireTimeOnChannel = clockTimeAtAfterCall
+			} else {
+				expectedFireTimeOnChannel = clockTimeAtAfterCall.Add(tc.afterDuration)
+			}
+
+			assertReceivesTime(t, ch, expectedFireTimeOnChannel, fireTestTimeout,
+				"Expected timer to fire with time %v", expectedFireTimeOnChannel)
+		})
+	}
+}
+
+func TestSimulatedClock_After_ShouldNotFire(t *testing.T) {
+	testCases := []struct {
+		name          string
+		afterDuration time.Duration
+		action        func(sc *SimulatedClock) // Action to manipulate the clock after After() is called
+	}{
+		{
+			name:          "PositiveDuration_DoesNotFire_WhenTimeAdvancedBeforeDuration",
+			afterDuration: 10 * time.Second,
+			action: func(sc *SimulatedClock) {
+				sc.AdvanceTime(5 * time.Second) // Advance, but not enough
+			},
+		},
+		{
+			name:          "PositiveDuration_DoesNotFire_WhenTimeSetBeforeDuration",
+			afterDuration: 10 * time.Second,
+			action: func(sc *SimulatedClock) {
+				sc.SetTime(referenceTime.Add(5 * time.Second)) // Set time, but not enough
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			clock := NewSimulatedClock(referenceTime)
+
+			ch := clock.After(tc.afterDuration)
+			require.NotNil(t, ch, "Channel should not be nil")
+
+			// Perform the action (if any) that might trigger the timer
+			if tc.action != nil {
```

Comment:
Let's remove this check if possible - I don't see this function being set to nil and the lesser the number of conditionals in test code, the better it is.